providers/implementations/keymgmt/template_kmgmt.c

```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core_dispatch.h>
+#include <openssl/core_names.h>
+#include <openssl/params.h>
+#include <openssl/err.h>
+#include <openssl/proverr.h>
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+#include <openssl/self_test.h>
+#include "internal/param_build_set.h"
+#include <openssl/param_build.h>
+#include "prov/implementations.h"
+#include "prov/providercommon.h"
+#include "prov/provider_ctx.h"
+#include "prov/securitycheck.h"
+
+extern const OSSL_DISPATCH ossl_template_keymgmt_functions[];
+
+#if defined(NDEBUG) || defined(OPENSSL_NO_STDIO)
+static void debug_print(char *fmt, ...)
+{
+}
+#else
+static void debug_print(char *fmt, ...)
+{
+    char out[1000];
+    va_list argptr;
+
+    va_start(argptr, fmt);
+    vsprintf(out, fmt, argptr);
+    va_end(argptr);
+    if (getenv("TEMPLATEKM"))
+        fprintf(stderr, "TEMPLATE_KM: %s", out);
+}
+#endif
+
+static OSSL_FUNC_keymgmt_new_fn template_new;
+static OSSL_FUNC_keymgmt_free_fn template_free;
+static OSSL_FUNC_keymgmt_gen_init_fn template_gen_init;
+static OSSL_FUNC_keymgmt_gen_fn template_gen;
+static OSSL_FUNC_keymgmt_gen_cleanup_fn template_gen_cleanup;
+static OSSL_FUNC_keymgmt_gen_set_params_fn template_gen_set_params;
+static OSSL_FUNC_keymgmt_gen_settable_params_fn template_gen_settable_params;
+static OSSL_FUNC_keymgmt_get_params_fn template_get_params;
+static OSSL_FUNC_keymgmt_gettable_params_fn template_gettable_params;
+static OSSL_FUNC_keymgmt_set_params_fn template_set_params;
+static OSSL_FUNC_keymgmt_settable_params_fn template_settable_params;
+static OSSL_FUNC_keymgmt_has_fn template_has;
+static OSSL_FUNC_keymgmt_match_fn template_match;
+static OSSL_FUNC_keymgmt_import_fn template_import;
+static OSSL_FUNC_keymgmt_export_fn template_export;
+static OSSL_FUNC_keymgmt_import_types_fn template_imexport_types;
+static OSSL_FUNC_keymgmt_export_types_fn template_imexport_types;
+
```

Comment:
Do any new key types have properties that can be validated for the public, private or keypair?

static OSSL_FUNC_keymgmt_validate_fn template_validate;

Reply:
Absolutely. But as this is key(type) specific, no such code can be added to a template. If your proposal is to add a comment in this regard (?), I surely could add that.

The "validate" function is intentionally left out as it's not required for enabling TLS1.3 KEM operation of a certain new algorithm class... :-) But any contributions to extend/complete these skeletons beyond this purpose are very much welcome!